File: 338-counting-bits/csharp/338-counting-bits.cs
public class Solution {
    public int[] CountBits(int num) {
        if (num == 0) return new int[] { 0 };
        var result = new int[num + 1];
        int power = 1;
        for (var i = 1; i <= num; ++i) {
            if (i == power * 2) power *= 2;
            result[i] = result[i - power] + 1;
        }
        return result;
    }
}

File: 338-counting-bits/python3/338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
      ans = [0] * (n+1)
      for i in range(len(ans)):
        x = 1
        while x <= i:
          if (x & i) > 0:
            ans[i] += 1
          x = x << 1
      return ans